### PR TITLE
feat(pgwire): support VARCHAR[] bind variables for symbol/varchar/str IN expressions

### PR DESCRIPTION
**Pull request by kafka1991**
Support VARCHAR array bind variables in PGWire, enabling queries like:

  ```java
  PreparedStatement stmt = conn.prepareStatement(
      "SELECT * FROM t WHERE col IN (?)"
  );
  stmt.setArray(1, conn.createArrayOf("varchar", new String[]{"a", "b", "c"}));
```

  **Supported**: SYMBOL/STRING/VARCHAR IN (?) with VARCHAR[] parameter.

  **Benefit**: Reuse prepared statements with different value lists, reducing query compilation overhead.

### CODE REVIEW

**Comment by glasstiger** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 277 / 307 (90.23%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/arr/FlatArrayView.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/eq/EqTimestampCursorFunctionFactory.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGOids.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java|46|56|82.14%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGNonNullVarcharArrayView.java|60|71|84.51%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGPipelineEntry.java|26|28|92.86%|
|:large_blue_circle:|io/questdb/cairo/arr/ArrayTypeDriver.java|45|47|95.74%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/QuaternaryFunction.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/UnaryFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InStrVarcharArrayFunctionFactory.java|33|33|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bind/ArrayBindVariable.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolCursorFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/lt/GtTimestampCursorFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/BinaryFunction.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/TernaryFunction.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/regex/AbstractLikeSymbolFunctionFactory.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/lt/LtTimestampCursorFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InVarcharVarcharArrayFunctionFactory.java|32|32|100.00%|
|:large_blue_circle:|io/questdb/cairo/arr/ArrayView.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/MultiArgFunction.java|2|2|100.00%|

**Comment by glasstiger** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 585 / 684 (85.53%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/arr/FlatArrayView.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/eq/EqTimestampCursorFunctionFactory.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bind/ArrayBindVariable.java|5|12|41.67%|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGPipelineEntry.java|33|53|62.26%|
|:large_blue_circle:|io/questdb/std/str/Utf8StringSinkList.java|51|67|76.12%|
|:large_blue_circle:|io/questdb/std/str/DirectUtf8StringList.java|18|22|81.82%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java|46|56|82.14%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGNonNullVarcharArrayView.java|60|71|84.51%|
|:large_blue_circle:|io/questdb/cairo/arr/VarcharArrayParser.java|187|210|89.05%|
|:large_blue_circle:|io/questdb/cairo/arr/ArrayTypeDriver.java|45|47|95.74%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InStrFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/QuaternaryFunction.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/UnaryFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InStrVarcharArrayFunctionFactory.java|35|35|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolCursorFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/lt/GtTimestampCursorFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/BinaryFunction.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/TernaryFunction.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/regex/AbstractLikeSymbolFunctionFactory.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/lt/LtTimestampCursorFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InVarcharVarcharArrayFunctionFactory.java|32|32|100.00%|
|:large_blue_circle:|io/questdb/std/DirectLongList.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InVarcharFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/arr/ArrayView.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGOids.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/MultiArgFunction.java|2|2|100.00%|

**Comment by questdb-butler** [comment]
## ⚠️ Enterprise CI Failed

The enterprise test suite failed for this PR.

**Build:** [View Details](https://dev.azure.com/questdb/questdb-enterprise/_build/results?buildId=205156)
**Tested Commit:** `92c6de87c37b4729fa59cfa03362bd67f34e4f6e`

Please investigate the failure before merging.

File: core/src/test/java/io/questdb/test/cutlass/pgwire/PGArraysTest.java [hunk 1]
```diff
@@ -881,6 +880,10 @@ private void assertArrayBindVarQueryFails(Connection connection, String query, S
         }
     }
 
+    private void assertArrayBindVarQueryFails1() {
```
Comment (bluestreak01) [inline core/src/test/java/io/questdb/test/cutlass/pgwire/PGArraysTest.java]: debris?

File: core/src/main/java/io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                    } else if (value.isAscii()) {
+                        intSet.add(symbolTable.keyOf(value.asAsciiCharSequence()));
+                    } else {
+                        StringSink sink = Misc.getThreadLocalSink();
+                        sink.clear();
+                        sink.put(value);
+                        intSet.add(symbolTable.keyOf(sink));
+                    }
+                }
+                testFunc = intTest;
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java]: we must not use lambda here, it is better to duplicate the code

File: core/src/test/java/io/questdb/test/cutlass/pgwire/PGArraysTest.java (hunk 1 above)
Comment (kafka1991) [reply core/src/test/java/io/questdb/test/cutlass/pgwire/PGArraysTest.java]: removed

File: core/src/main/java/io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java (hunk 2 above)
Comment (kafka1991) [reply core/src/main/java/io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java]: To me, Branching and using a lambda show the same performance in the benchmark.
We can’t completely eliminate the branch anyway, because the branch condition is decided in `Func.init`.
Change to use Branching

**Comment by glasstiger** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 584 / 681 (85.76%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/arr/FlatArrayView.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/eq/EqTimestampCursorFunctionFactory.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bind/ArrayBindVariable.java|5|12|41.67%|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGPipelineEntry.java|33|53|62.26%|
|:large_blue_circle:|io/questdb/std/str/Utf8StringSinkList.java|51|67|76.12%|
|:large_blue_circle:|io/questdb/std/str/DirectUtf8StringList.java|18|22|81.82%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java|45|53|84.91%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGNonNullVarcharArrayView.java|60|71|84.51%|
|:large_blue_circle:|io/questdb/cairo/arr/VarcharArrayParser.java|187|210|89.05%|
|:large_blue_circle:|io/questdb/cairo/arr/ArrayTypeDriver.java|45|47|95.74%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InStrFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/QuaternaryFunction.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/UnaryFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InStrVarcharArrayFunctionFactory.java|35|35|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolCursorFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/lt/GtTimestampCursorFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/BinaryFunction.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/TernaryFunction.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/regex/AbstractLikeSymbolFunctionFactory.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/lt/LtTimestampCursorFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InVarcharVarcharArrayFunctionFactory.java|32|32|100.00%|
|:large_blue_circle:|io/questdb/std/DirectLongList.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InVarcharFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/arr/ArrayView.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGOids.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/MultiArgFunction.java|2|2|100.00%|

File: core/src/main/java/io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+            if (intSet == null) {
+                intSet = new IntHashSet();
+            }
+            if (strSet == null) {
+                strSet = new CharSequenceHashSet();
+            }
+            intSet.clear();
+            strSet.clear();
+            ArrayView arrayView = arrayFunc.getArray(null);
+            StaticSymbolTable symbolTable = symbolFunc.getStaticSymbolTable();
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java:120]: it is worth exploring is this decision can be made before constructing the cursor. Left size of the "in" is non-variable. Deciding before the cursor creation will both simplify the logic and would remove branch in `getBool`

File: core/src/main/java/io/questdb/cairo/arr/ArrayView.java [hunk 4]
```diff
@@ -463,6 +464,10 @@ public final long getVanillaMemoryLayoutSize() {
                 + getCardinality() * elemSize; // data
     }
 
+    public final Utf8Sequence getVarchar(int flatIndex) {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cairo/arr/ArrayView.java:477]: lets document semantics of this method, if it returns mutable/reusable object or objects that can be cached externally. Otherwise this is ambiguous and we're continuing to create strings

File: core/src/main/java/io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+                        StringSink sink = Misc.getThreadLocalSink();
+                        sink.put(value);
+                        intSet.add(symbolTable.keyOf(sink));
+                    }
+                }
+                useIntSet = true;
+            } else {
+                for (int i = 0, n = arrayView.getCardinality(); i < n; i++) {
+                    Utf8Sequence element = arrayView.getVarchar(i);
+                    strSet.add(Utf8s.toString(element));
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java]: we should optimise for ascii chars. I suspect we can avoid string creation here altogether

File: core/src/main/java/io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+import io.questdb.std.str.Utf8Sequence;
+import io.questdb.std.str.Utf8s;
+
+/**
+ * Handles "symbol IN $1" where $1 is a VARCHAR[] bind variable.
+ * <p>
+ * VARCHAR[] is currently only supported as a bind variable type in PGWire protocol,
+ * passed via the BIND message. There is no SQL literal syntax for VARCHAR arrays.
+ */
+public class InSymbolVarcharArrayFunctionFactory implements FunctionFactory {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java:56]: isThreadSafe() behavior is dangling - pls have a look at InSymbolFunctionFactory

File: core/src/test/java/io/questdb/test/cutlass/pgwire/PGVarcharArrayBindVariablesTest.java [hunk 7]
```diff
@@ -0,0 +1,465 @@
...
+                                    """
+                                            ts[TIMESTAMP],varchar_col[VARCHAR]
+                                            """,
+                                    new StringSink(),
+                                    resultSet
+                            );
+                        }
+
+                        stmt.clearParameters();
+                        Array array3 = connection.createArrayOf("VARCHAR", new String[]{"b", null});
```
Comment (bluestreak01) [inline core/src/test/java/io/questdb/test/cutlass/pgwire/PGVarcharArrayBindVariablesTest.java:518]: can we add tests that pass null array?

**Comment by glasstiger** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 595 / 703 (84.64%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/arr/FlatArrayView.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/eq/EqTimestampCursorFunctionFactory.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bind/ArrayBindVariable.java|5|12|41.67%|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGPipelineEntry.java|33|53|62.26%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java|51|70|72.86%|
|:large_blue_circle:|io/questdb/std/str/Utf8StringSinkList.java|51|67|76.12%|
|:large_blue_circle:|io/questdb/std/str/DirectUtf8StringList.java|18|22|81.82%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGNonNullVarcharArrayView.java|60|71|84.51%|
|:large_blue_circle:|io/questdb/cairo/arr/VarcharArrayParser.java|187|210|89.05%|
|:large_blue_circle:|io/questdb/cairo/arr/ArrayTypeDriver.java|45|47|95.74%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InStrFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/QuaternaryFunction.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/UnaryFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InStrVarcharArrayFunctionFactory.java|35|35|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolCursorFunctionFactory.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/lt/GtTimestampCursorFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/BinaryFunction.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/TernaryFunction.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/regex/AbstractLikeSymbolFunctionFactory.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/lt/LtTimestampCursorFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InVarcharVarcharArrayFunctionFactory.java|32|32|100.00%|
|:large_blue_circle:|io/questdb/std/DirectLongList.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InVarcharFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/arr/ArrayView.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGOids.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/MultiArgFunction.java|2|2|100.00%|

**Comment by glasstiger** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 601 / 720 (83.47%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/CharSequenceHashSet.java|0|10|00.00%|
|:large_blue_circle:|io/questdb/cairo/arr/FlatArrayView.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/eq/EqTimestampCursorFunctionFactory.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bind/ArrayBindVariable.java|5|12|41.67%|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGPipelineEntry.java|33|53|62.26%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java|54|74|72.97%|
|:large_blue_circle:|io/questdb/std/str/Utf8StringSinkList.java|51|67|76.12%|
|:large_blue_circle:|io/questdb/std/str/DirectUtf8StringList.java|18|22|81.82%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGNonNullVarcharArrayView.java|60|71|84.51%|
|:large_blue_circle:|io/questdb/cairo/arr/VarcharArrayParser.java|187|210|89.05%|
|:large_blue_circle:|io/questdb/cairo/arr/ArrayTypeDriver.java|45|47|95.74%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InStrFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/WhereClauseParser.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/QuaternaryFunction.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/UnaryFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InStrVarcharArrayFunctionFactory.java|35|35|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolCursorFunctionFactory.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/lt/GtTimestampCursorFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/BinaryFunction.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/TernaryFunction.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/regex/AbstractLikeSymbolFunctionFactory.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/lt/LtTimestampCursorFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InVarcharVarcharArrayFunctionFactory.java|32|32|100.00%|
|:large_blue_circle:|io/questdb/std/DirectLongList.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InVarcharFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/arr/ArrayView.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGOids.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/MultiArgFunction.java|2|2|100.00%|

**Comment by glasstiger** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 682 / 806 (84.62%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/CharSequenceHashSet.java|0|10|00.00%|
|:large_blue_circle:|io/questdb/cairo/arr/FlatArrayView.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/eq/EqTimestampCursorFunctionFactory.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bind/ArrayBindVariable.java|5|12|41.67%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGPipelineEntry.java|33|53|62.26%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameSequence.java|5|7|71.43%|
|:large_blue_circle:|io/questdb/cairo/pool/PoolConstants.java|5|7|71.43%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java|54|74|72.97%|
|:large_blue_circle:|io/questdb/std/str/Utf8StringSinkList.java|51|67|76.12%|
|:large_blue_circle:|io/questdb/std/str/DirectUtf8StringList.java|18|22|81.82%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGNonNullVarcharArrayView.java|60|71|84.51%|
|:large_blue_circle:|io/questdb/cairo/arr/VarcharArrayParser.java|187|210|89.05%|
|:large_blue_circle:|io/questdb/cairo/arr/ArrayTypeDriver.java|45|47|95.74%|
|:large_blue_circle:|io/questdb/griffin/engine/table/ReaderPoolRecordCursorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/AsyncFilteredNegativeLimitRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/AsyncFilteredRecordCursor.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InStrFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/WhereClauseParser.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/QuaternaryFunction.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/UnaryFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InStrVarcharArrayFunctionFactory.java|35|35|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InSymbolCursorFunctionFactory.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TxnScoreboardPoolV2.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/lt/GtTimestampCursorFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractMultiTenantPool.java|18|18|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WalWriterPool.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/SqlCompilerPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/BinaryFunction.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/TernaryFunction.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/AsyncFilteredRecordCursorFactory.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/TableMetadataPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/jit/CompiledFilter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/regex/AbstractLikeSymbolFunctionFactory.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/AsyncFilterUtils.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/lt/LtTimestampCursorFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InVarcharVarcharArrayFunctionFactory.java|32|32|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameReduceTask.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/std/DirectLongList.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/AsyncJitFilteredRecordCursorFactory.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/SequencerMetadataPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InVarcharFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/arr/ArrayView.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/ReaderPool.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGOids.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/MultiArgFunction.java|2|2|100.00%|
